src/uu/dd/src/datastructures.rs

```diff
@@ -0,0 +1,197 @@
+// This file is part of the uutils coreutils package.
+//
+// (c) Tyler Steele <tyler.steele@protonmail.com>
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+/* cspell:disable */
+
+use crate::conversion_tables::*;
+
+use std::error::Error;
+use std::time;
+
+pub struct ProgUpdate {
+    pub reads_complete: u64,
+    pub reads_partial: u64,
+    pub writes_complete: u64,
+    pub writes_partial: u64,
+    pub bytes_total: u128,
+    pub records_truncated: u32,
+    pub duration: time::Duration,
+}
+
+pub struct ReadStat {
+    pub reads_complete: u64,
+    pub reads_partial: u64,
+    pub records_truncated: u32,
+}
+impl std::ops::AddAssign for ReadStat {
+    fn add_assign(&mut self, other: Self) {
+        *self = Self {
+            reads_complete: self.reads_complete + other.reads_complete,
+            reads_partial: self.reads_partial + other.reads_partial,
+            records_truncated: self.records_truncated + other.records_truncated,
+        }
+    }
+}
+
+pub struct WriteStat {
+    pub writes_complete: u64,
+    pub writes_partial: u64,
+    pub bytes_total: u128,
+}
+impl std::ops::AddAssign for WriteStat {
+    fn add_assign(&mut self, other: Self) {
+        *self = Self {
+            writes_complete: self.writes_complete + other.writes_complete,
+            writes_partial: self.writes_partial + other.writes_partial,
+            bytes_total: self.bytes_total + other.bytes_total,
+        }
+    }
+}
+
+type Cbs = usize;
+
+/// Stores all Conv Flags that apply to the input
+pub struct IConvFlags {
+    pub ctable: Option<&'static ConversionTable>,
+    pub block: Option<Cbs>,
+    pub unblock: Option<Cbs>,
+    pub swab: bool,
+    pub sync: Option<u8>,
+    pub noerror: bool,
+}
+
+/// Stores all Conv Flags that apply to the output
+#[derive(Debug, PartialEq)]
+pub struct OConvFlags {
+    pub sparse: bool,
+    pub excl: bool,
+    pub nocreat: bool,
+    pub notrunc: bool,
+    pub fdatasync: bool,
+    pub fsync: bool,
+}
+
+/// Stores all Flags that apply to the input
+pub struct IFlags {
+    #[allow(dead_code)]
+    pub cio: bool,
+    #[allow(dead_code)]
+    pub direct: bool,
+    #[allow(dead_code)]
+    pub directory: bool,
+    #[allow(dead_code)]
+    pub dsync: bool,
+    #[allow(dead_code)]
+    pub sync: bool,
+    #[allow(dead_code)]
+    pub nocache: bool,
+    #[allow(dead_code)]
+    pub nonblock: bool,
+    #[allow(dead_code)]
+    pub noatime: bool,
+    #[allow(dead_code)]
+    pub noctty: bool,
+    #[allow(dead_code)]
+    pub nofollow: bool,
+    #[allow(dead_code)]
+    pub nolinks: bool,
+    #[allow(dead_code)]
+    pub binary: bool,
+    #[allow(dead_code)]
+    pub text: bool,
```

Comment:
Are these still dead code?

Reply:
Or are they linux only? In that case they should have an attribute that reflects that (i.e. `#[cfg(target_os = "linux")]`)